src/fiat/mpl/internal/mpl_allreduce_mod.F90

```diff
@@ -505,29 +512,21 @@ SUBROUTINE MPL_ALLREDUCE_REAL8_2D(PSENDBUF,CDOPER,LDREPROD, &
   IF (PRESENT(LDREPROD)) THEN
     LLREPRODSUM = LDREPROD
   ELSE
-    CALL MPL_MESSAGE(IERROR,&
-     & 'MPL_ALLREDUCE: SUMMATION OPERATOR NOT REPRODUCIBLE IN REAL MODE',&
-     & CDSTRING,LDABORT=LLABORT)
+    CALL MPL_MESSAGE(CDMESSAGE='MPL_ALLREDUCE: SUMMATION OPERATOR NOT REPRODUCIBLE IN REAL MODE',&
+     & CDSTRING=CDSTRING,KERROR=IERROR,LDABORT=LLABORT)
   ENDIF
 ELSE
-  CALL MPL_MESSAGE(IERROR,'MPL_ALLREDUCE: ERROR UNKNOWN OPERATOR',&
-   & CDSTRING,LDABORT=LLABORT)
+  CALL MPL_MESSAGE(CDMESSAGE='MPL_ALLREDUCE: ERROR UNKNOWN OPERATOR',&
+   & CDSTRING=CDSTRING,KERROR=IERROR,LDABORT=LLABORT)
 ENDIF
 
 IF(PRESENT(KCOMM)) THEN
-  ICOMM=KCOMM
+  ICOMM%MPI_VAL=KCOMM
 ELSE
-  ICOMM=MPL_COMM_OML(ITID)
+  ICOMM%MPI_VAL=MPL_COMM_OML(ITID)
 ENDIF
 
 ISENDCOUNT = SIZE(PSENDBUF)
-!#ifndef NAGFOR
-!IF (ISENDCOUNT > 0) THEN
-!  IF( (LOC(PSENDBUF(UBOUND(PSENDBUF,1)))-LOC(PSENDBUF(LBOUND(PSENDBUF,1)))) /= 8_JPIB*(ISENDCOUNT - 1) ) THEN
-!    CALL MPL_MESSAGE(CDMESSAGE='MPL_ALLREDUCE: BUFFER NOT CONTIGUOUS ',LDABORT=LLABORT)
-!  ENDIF
-!ENDIF
-!#endif
```

Comment:
Not required?

Reply:
This was commented out upstream, so I deleted it, but it might indeed make more sense to make it homogenous with other subroutines.

Reply:
Right did not notice it was already commented, and saw it as a delete only.